elyra/pipeline/component.py

```diff
@@ -136,6 +140,18 @@ def default_control_type(self) -> str:
         """
         return self._default_control_type
 
+    @property
+    def default_data_type(self) -> str:
+        """
+            The `default_data_type` is the first data type that is assigned to this specific parameter
+            after parsing the component specification.
+        """
+        return self._default_data_type
+
+    @property
+    def allow_no_options(self) -> str:
+        return "true" if self._allow_no_options else "false"
```

Comment:
Having a member variable property masquerade as a different data type just doesn't strike me as a good practice.

Can the boolean value be coerced to the appropriate string within the template - since that's the only time it's needed as a string?  If not, then I would recommend renaming the property method to something like `allow_no_options_to_jinja`, or pass a boolean to the property method indicating you want the lowercase string form and make the return type of the `allow_no_options` property a `Union[bool, str]` with the default behavior that it returns its native data type.

Reply:
:nods in agreement: 
Updated!